google/cloud/aiplatform/metadata/artifact.py

```
@@ -0,0 +1,285 @@
+                the format:
+                projects/{project}/locations/{location}/metadataStores/{metadata_store_id}/artifacts/{artifact_id}.
+            metadata (Dict):
+                Optional. metadata information to update the artifact with.
+            metadata_store_id (str):
+                The {metadata_store_id} portion of the resource name with
+                the format:
+                projects/{project}/locations/{location}/metadataStores/{metadata_store_id}/artifacts/{artifact_id}
+                If not provided, the MetadataStore's ID will be set to "default".
+            project (str):
+                Project where this artifact belongs. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Location where this artifact belongs. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to update this artifact. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            artifact (_Artifact):
+                Updated representation of the managed Metadata Artifact resource.
+
+        """
+
+        gapic_artifact = cls(
+            artifact_name=artifact_id,
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )._gca_resource
+        gapic_artifact.metadata = metadata
+
+        resource_name = super().update(
+            resource_id=artifact_id,
+            resource_noun=cls._resource_noun,
+            gapic_resource=gapic_artifact,
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        if not resource_name:
+            raise ValueError("Error while updating artifact")
```

It's better to propagate the service error.

---

Done! thanks!